plugins/@grouparoo/mailchimp/src/lib/shared/updateProfile.ts

```diff
@@ -34,12 +40,19 @@ export const updateProfile: UpdateProfileMethod = async ({
   if (!mailchimpId) {
     throw new Error("mailchimpId is required");
   }
-
   if (toDelete) {
     if (noDelete) {
       return { success: true };
     }
-    await deleteMember(client, listId, mailchimpId, oldGroups, newGroups);
+    await deleteContactOrClearGroups(
+      client,
+      listId,
+      mailchimpId,
```

Comment:
What happens if there's an email change _and_ a `toDelete`? The code in exportProfile that deals with changing IDs only runs if `toDelete` is `false`, which means that when you get to this point you might be deleting based on the new email instead of the old one.